frame/staking/src/pallet/impls.rs

```diff
@@ -775,10 +775,8 @@ impl<T: Config> Pallet<T> {
 	pub fn do_add_nominator(who: &T::AccountId, nominations: Nominations<T::AccountId>) {
 		if !Nominators::<T>::contains_key(who) {
 			// maybe update sorted list. Error checking is defensive-only - this should never fail.
-			if T::SortedListProvider::on_insert(who.clone(), Self::weight_of(who)).is_err() {
-				log!(warn, "attempt to insert duplicate nominator ({:#?})", who);
-				debug_assert!(false, "attempt to insert duplicate nominator");
-			};
+			let _ = T::SortedListProvider::on_insert(who.clone(), Self::weight_of(who))
```

Comment:
In the future we could add a `defensive_assert_ok` or something like that, so that you don't need to ignore the return value here.

Reply:
I like the idea, but to me `assert` means this will panic, whereas here it won't.

For now I'd say let's keep the traits and API minimal for now and see what things we actually need in action. Adding more `defensive_xxx` is quite easy from this point onwards.